neonvm-runner/cmd/main.go

```diff
@@ -851,6 +861,19 @@ func buildQEMUCmd(
 		qemuCmd = append(qemuCmd, "-drive", fmt.Sprintf("id=%s,file=%s,if=virtio,media=disk,%s,discard=unmap", swapName, dPath, cfg.diskCacheSettings))
 	}
 
+	if cfg.architecture == architectureArm64 {
+		// add custom firmware to have ACPI working
+		qemuCmd = append(qemuCmd, "-bios", "/vm/QEMU_EFI_ARM.fd")
+		// arm virt has only one UART, setup virtio-serial to add more /dev/hvcX
+		qemuCmd = append(qemuCmd,
+			"-chardev", "stdio,id=virtio-console",
+			"-device", "virtconsole,chardev=virtio-console",
+		)
+	} else {
+		// on amd we have multiple UART ports so we can just use serial stdio
+		qemuCmd = append(qemuCmd, "-serial", "stdio")
+	}
```

Comment:
suggestion: let's use `switch` for these and panic in the default case, so it's more clear which architecture gets which behavior?

(+ same suggestion for other occurrences of this)

Reply:
changed to switch